src/main/java/com/spotify/autoscaler/ClusterStats.java

```diff
@@ -211,62 +107,43 @@ void setStats(BigtableCluster cluster, int currentNodes) {
-              .tagged("project-id", cluster.projectId())
-              .tagged("cluster-id", cluster.clusterId())
-              .tagged("instance-id", cluster.instanceId()),
-          (Gauge<Integer>) () -> registeredClusters.get(cluster.clusterName()).getMinNodeCount());
-
-      this.registry.register(
-          APP_PREFIX
-              .tagged("what", "max-node-count")
-              .tagged("project-id", cluster.projectId())
-              .tagged("cluster-id", cluster.clusterId())
-              .tagged("instance-id", cluster.instanceId()),
-          (Gauge<Integer>) () -> registeredClusters.get(cluster.clusterName()).getMaxNodeCount());
-
-      this.registry.register(
-          APP_PREFIX
-              .tagged("what", "effective-min-node-count")
-              .tagged("project-id", cluster.projectId())
-              .tagged("cluster-id", cluster.clusterId())
-              .tagged("instance-id", cluster.instanceId()),
-          (Gauge<Integer>)
-              () -> registeredClusters.get(cluster.clusterName()).getEffectiveMinNodeCount());
-
-      this.registry.register(
-          APP_PREFIX
-              .tagged("what", "last-check-time")
-              .tagged("project-id", cluster.projectId())
-              .tagged("cluster-id", cluster.clusterId())
-              .tagged("instance-id", cluster.instanceId()),
-          (Gauge<Long>)
-              () ->
-                  db.getBigtableCluster(
-                          cluster.projectId(), cluster.instanceId(), cluster.clusterId())
-                      .flatMap(
-                          p ->
-                              Optional.of(
-                                  Duration.between(
-                                      p.lastCheck().orElse(Instant.EPOCH), Instant.now())))
-                      .get()
-                      .getSeconds());
+      for (BigtableMetric.Metrics metric : BigtableMetric.Metrics.values()) {
+        // This will return null for non implemented getMetricValues. This will happen when the
+        // metric depends in other things other than the registeredClusters or the cluster
+        // itself, e.g., when it is dependent on the database.
+        Gauge metricValue = metric.getMetricValue(registeredClusters.get(cluster.clusterName()));
+        switch (metric) {
+          case LAST_CHECK_TIME:
+            metricValue =
+                (Gauge<Long>)
+                    () ->
+                        db.getBigtableCluster(
+                                cluster.projectId(), cluster.instanceId(), cluster.clusterId())
+                            .flatMap(
+                                p ->
+                                    Optional.of(
+                                        Duration.between(
+                                            p.lastCheck().orElse(Instant.EPOCH), Instant.now())))
+                            .get()
+                            .getSeconds();
+            break;
+          case CPU_TARGET_RATIO:
```

Comment:
this can be 
`return () -> clusterData.getCpuUtil() / clusterData.getCluster().cpuTarget();` in the enum

Reply:
👍